Remove dead code: byte_extract_(big|little)_endian

allredj opened the pull request:
Constructors for `byte_extract_big_endian` and `byte_extract_little_endian` are not used anywhere. I suggest removing them, because their presence is a bit confusing (endianness is actually set using `byte_extract_id()` and `byte_update_id()`).

Some other methods become redundant.

Any thoughts?

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 869faf52dbaeecbf1d4e86d6dc4dc92d5bddcfd2).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77689273

tautschnig reviewed (approved):
I'm always in favour of removing code :-)  I think those once seemed like a good idea, but really if by now we don't use them then probably we never will. I'd suggest @kroening for final approval, though.

Commit:
Remove dead code: byte_extract_(big|little)_endian

Constructors for byte_extract_big_endian and byte_extract_little_endian
are not used.

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: b18b0314167ba0adf932008e8cc5be958f47687e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88985499
